File: chapter1.2 find element/B1041 test num.cpp
#include <cstdio>
const int N = 1010;
struct stu{
	long long num;
	int real;
}test[N];

int main(){
	int n;
	scanf("%d", &n);
	int b, c;
	long long a;
	for (int i = 0; i < n; ++i){
		scanf("%lld%d%d", &a, &b, &c);
		test[b].num = a;
		test[b].real = c;
	}
	int m;
	scanf("%d", &m);
	int temp;
	for (int i = 0; i < m; ++i){
		scanf("%d", &temp);
		printf("%lld %d\n", test[temp].num, test[temp].real);
	}
}

//用结构体把关系对应起来就十分明了

File: chapter1.2 find element/B1041 test num.py
n = int(raw_input())
num = [0] * (n + 1)
test = [0] * (n + 1)
real = [0] * (n + 1)
for i in range(1, n + 1):
	num[i], test[i], real[i] = map(int, raw_input().split())
m = int(raw_input())
temp = [0] * n
temp = map(int, raw_input().split())
for i in range(m):
	for j in range(1,n+1):
		if test[j] == temp[i]:
			print num[j], real[j]
			
#虽然答案正确，通过case
#就是一个最简单的查询，但是这样没有建立一个对应关系，低效
#直接用C中的结构体更简单